raiden/tests/integration/network/transport/rif_comms/test_client.py

```diff
@@ -23,92 +24,173 @@ def comms_nodes(amount_of_nodes) -> {int, CommsNode}:
 
 
 @pytest.mark.parametrize("amount_of_nodes", [1])
-def test_locate_own_peer_id(comms_nodes):
-    comms_node = comms_nodes[1]
+def test_subscribe_to_invalid(comms_nodes):
+    client = comms_nodes[1].client
```

Comment:
I don't like the index access that we need to do for testing in general, anyway i don't know how to avoid that neither

Reply:
> I don't like the index access that we need to do for testing in general, anyway i don't know how to avoid that neither

what do you not like about it? we are receiving comms nodes through the fixture, in the form of a collection.

how would you prefer to access this collection? by name? something like `comms_nodes["one"]`?

Reply:
i mean i know this is how it works but it looks weird like saying something like:
```python
@pytest.mark.parametrize("amount_of_nodes", [1])
def test_locate_own_peer_id(comms_nodes):
    comms_nodes[0]
...
```
Why? because if you are saying that comms_nodes is 1 it's better to have something like:
```python
def test_locate_own_peer_id(comms_node):
comms_node.something
```
Anyway i know this is part of the language or the framework that we are using, but i just don't like to access lists by index in any language

Reply:
> i mean i know this is how it works but it looks weird like saying something like:
> 
> ```python
> @pytest.mark.parametrize("amount_of_nodes", [1])
> def test_locate_own_peer_id(comms_nodes):
>     comms_nodes[0]
> ...
> ```
> 
> Why? because if you are saying that comms_nodes is length 1 it's better to have something like:
> 
> ```python
> def test_locate_own_peer_id(comms_node):
> comms_node.something
> ```
> 
> Anyway i know this is part of the language or the framework that we are using, but i just don't like to access lists by index in any language

i disagree only due to the fact that i don't like the readability of this:

```python
client_1 = comms_node[0]
```

and prefer the way it currently is:
```python
client_1 = comms_node[1]
```

but i appreciate the sentiment regarding list access by index. as you say, it is unavoidable for now.